Source/WebCore/rendering/svg/RenderSVGImage.cpp

```diff
@@ -308,9 +308,10 @@ void RenderSVGImage::repaintOrMarkForLayout(const IntRect* rect)
 
     FloatRect repaintRect = borderBoxRectEquivalent();
     if (rect) {
-        // The image changed rect is in source image coordinates (pre-zooming),
+        // The image changed rect is in source image coordinates (without zoom),
         // so map from the bounds of the image to the contentsBox.
-        repaintRect.intersect(enclosingIntRect(mapRect(*rect, FloatRect(FloatPoint(), imageResource().imageSize(1.0f)), repaintRect)));
+        const LayoutSize imageSizeWithoutZoom = m_imageResource->imageSize(1 / style().effectiveZoom());
+        repaintRect.intersect(enclosingIntRect(mapRect(*rect, FloatRect(FloatPoint(), imageSizeWithoutZoom), repaintRect)));
```

Comment:
RenderSVGImage is only used for LBSE. Is your plan not to fix it for LegacyRenderSVGImage? Also RenderSVGImage and LegacyRenderSVGImage are created for SVG `<image>` element https://developer.mozilla.org/en-US/docs/Web/SVG/Element/image. Your test case does not include any SVG `<image>` element. Instead it includes the HTML `<img>` element which is rendered through SVGImage::draw().

Reply:
Sure - I will add similar to LegacyRenderSVGImage. As for the test case, I am doing merge from Blink patch and it is more repaint / style invalidation fix when CSS animations are applied to SVG image in `<img>` rather than `<image>` element while zoomed. If you want me to add `<image>` case then I can try but I just started dwelling in Webkit Codebase and still novice so I might not be able to do it for sure but worth giving it a try.

Reply:
I just had a look into LegacyRenderSVGImage and I think it does not have same codepath as the above and in some point from Legacy to LBSE, we moved this part to LBSE use and I think this Blink merge would be limited to this path. If I am missing anything, happy to be guided.

Reply:
The reported test case in Bugzilla https://jstsch.com/misc/2019/7/safari-svg-zoom.html is not fixed with this change. This happens with LBSE on or off.

Also when opening the test case you added in this PR in mini-browser, it is not displayed correctly with this change. A pink rectangle is drawn with smaller green rectangle at its top-left corner. When refreshing the page only a green rectangle is displayed. This happens with LBSE on and off.

So I am not sure what this PR is fixing. I think you need to investigate these two cases and understand the problem more. Feel free to ask questions on slack. Also make sure the added layout test exercises the modified code.

Reply:
Closing PR to investigate and documenting stuff on bug as well.

Reply:
LBSE + zooming is currently broken anyhow -- a general fix is underways for LBSE that will fix zooming/panning for absolute/relative sized documents with/without viewBox, within <fO> subtrees, SVGImages, etc..
-> A huge interop benefit, as Safari currently deviates (impossible to enlarge SVGs with relative sized width/height that have a viewBox set) from all other browsers [funny thought that I. made it like this also to restore inter-op in 2012, where Opera/FF followed these rules).